components/inputs/select.tsx

```diff
@@ -22,16 +29,36 @@ function useSelect() {
 
   return context;
 }
+type Path =
+  | { navigation?: boolean; prefetch?: boolean; variant: 'ui' }
+  | { navigation?: never; prefetch?: never; variant: 'form' };
 
 const Select = ({
-  variant = 'ui',
+  navigation,
+  prefetch,
+  variant,
   ...props
-}: React.ComponentPropsWithoutRef<typeof SelectPrimitive.Root> & {
-  variant?: 'form' | 'ui';
-}) => {
+}: React.ComponentPropsWithoutRef<typeof SelectPrimitive.Root> & Path) => {
+  const router = useRouter();
+  const routerPrefetch = React.useCallback(
+    (path: string) => {
+      router.prefetch(path);
+    },
+    [router]
+  );
   return (
-    <SelectContext.Provider value={{ variant }}>
-      <SelectPrimitive.Root {...props} />
+    <SelectContext.Provider value={{ prefetch, routerPrefetch, variant }}>
+      {variant === 'ui' && navigation ? (
+        <SelectPrimitive.Root
+          {...props}
+          onValueChange={(path) => {
+            router.push(path);
+            props.onValueChange ? props.onValueChange(path) : null;
```

Comment:
Bad variable naming.
```suggestion
          onValueChange={(value) => {
            router.push(value);
            props.onValueChange ? props.onValueChange(value) : null;
```